gcc7 compilation warning(s) fix(es) in HLTrigger/Timer

mrodozov opened the pull request:
Fix for compilation warning(s) when compiling with gcc7 and more flags listed here:
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc700/CMSSW_9_3_X_2017-07-24-2300/HLTrigger/Timer

Commit:
gcc7 compilation warning(s) fix(es) in HLTrigger/Timer

cmsbuild commented:
A new Pull Request was created by  @mrodozov  for master.

It involves the following packages:

HLTrigger/Timer


@Martin-Grunewald, @silviodonato, @cmsbuild, @fwyzard, @davidlange6 can you please review it and eventually sign? Thanks.
@Martin-Grunewald this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

Martin-Grunewald commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

fwyzard commented on HLTrigger/Timer/test/chrono/src/x86_tsc.cc:
```diff
@@ -163,16 +163,19 @@ extern "C" {
 
   static uint64_t (*serialising_rdtsc_resolver(void))(void)
   {
+    
```
please do not add extra empty lines

fwyzard commented on HLTrigger/Timer/test/chrono/src/x86_tsc.cc:
```diff
@@ -163,16 +163,19 @@ extern "C" {
     if (has_tsc()) {
       // if the TSC is available, chck the processor vendor
-      unsigned int eax, ebx, ecx, edx;
+      //unsigned int eax, ebx, ecx, edx;
```
why move it out of the `if` block ?

fwyzard commented:
-1

fwyzard commented:
please change only line 175 to
```
unsigned int eax = 0, ebx = 0, ecx = 0, edx = 0;
```

cmsbuild commented:
+1
Tested at: 379440aa2e743cc7145e6f4442d58fcd6956ae90
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19906/21749/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/939e20418c614166893c04ce9dd9075961f3f229
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19906/21749/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19906/21749/git-merge-result

cmsbuild commented:
Comparison job queued.

Commit:
updating the file as requested

updating the file with keeping the vars inside the function and init them with zeros

cmsbuild commented:
Pull request #19906 was updated. @Martin-Grunewald, @silviodonato, @cmsbuild, @fwyzard, @davidlange6 can you please check and sign again.

fwyzard commented:
can you
  - not add the empty line 166
  - not add spaces at the end of line 172/173
  - squash all commits into a single one
?

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19906/21749/summary.html

Comparison Summary:
 - You potentially added 5 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 2363781
 - DQMHistoTests: Total failures: 29975
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2333640
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 93 log files, 14 edm output root files, 23 DQM output files

davidlange6 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

fwyzard commented:
-1

fwyzard commented:
see comments above

mrodozov commented:
I'm closing this pr find the new one here https://github.com/cms-sw/cmssw/pull/19919

cmsbuild commented:
+1
Tested at: d6c60e646a999c71cc636022bb3a6f288a22aa64
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19906/21768/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/cc98e017e9bba25fabfa2adfec24cc3cb0378556
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19906/21768/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19906/21768/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19906/21768/summary.html

Comparison Summary:
 - You potentially added 5 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 2363781
 - DQMHistoTests: Total failures: 15145
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2348470
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 93 log files, 14 edm output root files, 23 DQM output files
